Update docs with ansible/ansible-core version, AAP and fix the dated git issue templates

ddimatos opened the pull request:
##### SUMMARY
Addresses issue #770 

##### ISSUE TYPE
- Docs Pull Request

ddimatos commented:
![image](https://github.com/ansible-collections/ibm_zos_core/assets/25803172/fb5039fd-07b3-40b0-a4b1-d5cc4c6bbc07)

ddimatos commented:
![image](https://github.com/ansible-collections/ibm_zos_core/assets/25803172/f9e93b4f-bc74-4b11-8834-087d503606ba)

ddimatos commented:
I see no reason to run regression, no source was changed.
On another note, this does speak to ibm_zos_core version 1.5.0 but we have no maintenance branch started and don't know if its needed to start one based on a doc change. 

We can't update the README in Galaxy without a 1.5.1 release thus see no point to start a maintenance branch until we have module bug. The general docs https://ibm.github.io/z_ansible_collections_doc/index.html will be updated with the release notes one this merges.

AndreMarcel99 commented on .github/ISSUE_TEMPLATE/doc_issue.yml:
```diff
@@ -40,20 +40,23 @@ body:
+        - v1.9.0
+        - v1.8.0
+        - v1.7.0
+        - v1.6.0
```
I prefer the new order from newest to oldest by descending. Just two personal doubts
Before were include the betas version, wouldn't be nice to keep with that including the 1.6.0-beta at the moment?
Also Is ok to add all that next versions?
The second doubt apply to the the 3 files changes in ISSUE_TEMPLATE.

ddimatos replied:
@AndreMarcel99 - I think what you prefer the descending (newest to oldest) is what how it will show in the git issue, this PR is actually correcting the behavior that you are commenting on. GH issues show the selection box as it is ordered, have a look at [here](https://raw.githubusercontent.com/ansible-collections/ibm_zos_core/dev/.github/ISSUE_TEMPLATE/doc_issue.yml) and you will see the order starts with 1.0 to later versions, so we are both in agreement, and this is aligning to your comment.

ddimatos replied:
This template is for documentation, our docs really don't align to a version (we have work items to have version support on our docs in the backlog in JIRA (not yet ported to GH), I figured it might be good to have a version to associate to a doc , maybe its better we do use a `x` for the last bit , eg 1.6.x. Our docs are slightly different where with issues related to modules and source we actually do need to include beta's. I could push this out with `x`'s instead.

ddimatos replied:
I think its ok to add all the next versions, its more or less future proofing, if we find users are selecting them when there is no version we can remove them, what I find worse that happens is we are not keeping up with updates to the templates hence I added a few more future release versions. Not sure what is worse, not having the templates updated with the most current release or having future releases, I tend to think its worse to not have the templates updated so when someone tries to open an issue, the version is not there.

ddimatos replied:
I suppose another option is as part of our release strategy we update the templates, that could solve this problem.

AndreMarcel99 reviewed (approved):
Just two personal doubts
Before were include the betas version, wouldn't be nice to keep with that including the 1.6.0-beta at the moment?
Also Is ok to add all that next versions?
The second doubt apply to the the 3 files changes in ISSUE_TEMPLATE.

ddimatos commented:
> Just two personal doubts Before were include the betas version, wouldn't be nice to keep with that including the 1.6.0-beta at the moment? Also Is ok to add all that next versions? The second doubt apply to the the 3 files changes in ISSUE_TEMPLATE.

Good points @AndreMarcel99 - i will add the beta's back to the bug templates, i think copy and paste was to easy :) and overlooked that. I meant to keep the 1.6.0-beta.1 and 1.7.0-beta.1 and upwards , but older we don't need beta's after they GA. I'll push some changes.

ddimatos commented:
@AndreMarcel99 - I updated the approach and added the beta's to all, the order should be descending even thought it appears not, I also only included the latest of each release, this is based on the idea if a bug is opened say on 1.4.0 and we have 1.4.1, we should ask the user to be on the latest of that version should we need to perform a fix.
eg, 
```
        - v1.9.0
        - v1.9.0-beta.1
        - v1.8.0
        - v1.8.0-beta.1
        - v1.7.0
        - v1.7.0-beta.1
        - v1.6.0
        - v1.6.0-beta.1
        - v1.5.0
        - v1.4.1
        - v1.3.6
        - v1.2.1
        - v1.1.0
        - v1.0.0
```
